PacketDotNet/DrdaDDMPacket.cs

```diff
@@ -0,0 +1,216 @@
+        public byte Format
+        {
+            get
+            {
+                return header.Bytes[header.Offset + DrdaDDMFields.FormatPosition];
+            }
+        }
+
+        /// <summary>
+        /// The CorrelId field
+        /// </summary>
+        public ushort CorrelId
+        {
+            get
+            {
+                return BigEndianBitConverter.Big.ToUInt16(header.Bytes, header.Offset + DrdaDDMFields.CorrelIdPosition);
+            }
+        }
+
+        /// <summary>
+        /// The Length2 field
+        /// </summary>
+        public ushort Length2
+        {
+            get
+            {
+                return BigEndianBitConverter.Big.ToUInt16(header.Bytes, header.Offset + DrdaDDMFields.Length2Position);
+            }
+        }
+
+        /// <summary>
+        /// The Code Point field
+        /// </summary>
+        public DrdaCodepointType CodePoint
+        {
+            get
+            {
+                return (DrdaCodepointType)BigEndianBitConverter.Big.ToUInt16(header.Bytes, header.Offset + DrdaDDMFields.CodePointPosition);
+            }
+        }
+
+        private List<DrdaDDMParameter> paramters;
+        public List<DrdaDDMParameter> Parameters
+        {
+            get
+            {
+                if (paramters == null) paramters = new List<DrdaDDMParameter>();
+                if (paramters.Count > 0) return paramters;
+                var offset = header.Offset + DrdaDDMFields.DDMHeadTotalLength;
+                while (offset < header.BytesLength)
+                {
+                    int length = BigEndianBitConverter.Big.ToUInt16(header.Bytes, offset);
+                    if (length == 0)
+                    {
+                        length = header.BytesLength - offset;
+                    }
+                    if (offset + length <= header.BytesLength)
+                    {
+                        var parameter = new DrdaDDMParameter()
+                        {
+                            Length = length,
+                            DrdaCodepoint = (DrdaCodepointType)BigEndianBitConverter.Big.ToUInt16(header.Bytes, offset + DrdaDDMFields.ParameterLengthLength)
+                        };
+                        
+                        if(parameter .DrdaCodepoint== DrdaCodepointType.DATA|| parameter.DrdaCodepoint == DrdaCodepointType.QRYDTA)
+                        {
+                            var startIndex = offset + DrdaDDMFields.ParameterLengthLength + DrdaDDMFields.ParameterCodePointLength;
+                            var strLength = length - 4;
+                            parameter.Data = ASCIIEncoding.UTF8.GetString(header.Bytes, startIndex,strLength);
+                        }
+                        else
+                        {
+                            parameter.Data = StringConverter.EbcdicToAscii(header.Bytes, offset + DrdaDDMFields.ParameterLengthLength + DrdaDDMFields.ParameterCodePointLength, length - 2 - 2);
```

Comment:
And here with the -2 and -2. I presume those are accounting for something in the data fields so two variables, one for each 2, would help describe why the value is 2.